src/app/forms/components/date/date.component.ts

```diff
@@ -104,40 +109,55 @@ export class DateComponent extends BaseControlComponent implements OnInit, OnDes
   subscribeAndPropagateChanges() {
     let dateFields;
 
-    if(this.includeTime){
-      dateFields = { day: this.day$, month: this.month$, year: this.year$, hour: this.hour$, minute: this.minute$ }
-    }else{
-      dateFields = { day: this.day$, month: this.month$, year: this.year$ }
+    if (this.includeTime) {
+      dateFields = { day: this.day$, month: this.month$, year: this.year$, hour: this.hour$, minute: this.minute$ };
+    } else {
+      dateFields = { day: this.day$, month: this.month$, year: this.year$ };
     }
 
     return combineLatest(dateFields).subscribe({
-      
       next: ({ day, month, year, hour, minute }) => {
-        if (!day || !month || !year || (this.includeTime && (!hour || !minute))) {
+        if (!day && !month && !year && !hour && !minute) {
           this.onChange(null);
           return;
         }
 
-        const date = new Date(Date.UTC(year, month - 1, day));
-
-        hour = hour ?? new Date(this.originalDate).getHours();
-        minute = minute ?? new Date(this.originalDate).getMinutes();
-        const second = new Date(this.originalDate).getSeconds();
-
-        if ('Invalid Date' !== date.toString()) {
-          date.setHours(hour || 0);
-          date.setMinutes(minute || 0);
-          date.setSeconds(second || 0);
-        }
+        hour = this.includeTime ? hour : this.originalDate ? new Date(this.originalDate).getHours() : '00';
+        minute = this.includeTime ? minute : this.originalDate ? new Date(this.originalDate).getMinutes() : '00';
+        const second = this.originalDate ? new Date(this.originalDate).getSeconds() : '00';
 
-        this.onChange(date);
+        this.onChange(
+          `${year || ''}-${this.padded(month)}-${this.padded(day)}T${this.padded(hour)}:${this.padded(minute)}:${this.padded(second)}.000Z`
+        );
       }
     });
+  }
 
-  
+  padded(n: number | string | undefined, l = 2) {
+    const val = undefined !== n && null !== n ? String(n).padStart(l, '0') : 'NaN';
```

Comment:
Using a juggling-check, you can test both `null` and `undefined` in one hit:
```
if (x == null) {
```
So, in your case:
```suggestion
    const val = n != null ? String(n).padStart(l, '0') : 'NaN';

```